### PR TITLE
Add ProtT5-XL-BFD model card

### PR DESCRIPTION
**Pull request by agemagician**
Fixes # (issue)
Create a new card for our ProtT5-XL-BFD model.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dimiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests), 
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to the it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests? 

## Who can review?
Model Cards: @julien-c
T5: @patrickvonplaten

### CODE REVIEW

File: model_cards/Rostlab/prot_t5_xl_bfd/README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+# ProtT5-XL-BFD model
+
+Pretrained model on protein sequences using a masked language modeling (MLM) objective. It was introduced in
+[this paper](https://doi.org/10.1101/2020.07.12.199554) and first released in
+[this repository](https://github.com/agemagician/ProtTrans). This model is trained on uppercase amino acids: it only works with capital letter amino acids.
+
+
+## Model description
+
+ProtT5-XL-BFD is based on T5 model which pretrained on a large corpus of protein sequences in a self-supervised fashion.
```
Comment (patrickvonplaten) [inline model_cards/Rostlab/prot_t5_xl_bfd/README.md:18]:
```suggestion
ProtT5-XL-BFD is based on the T5 model that is pretrained on a large corpus of protein sequences in a self-supervised fashion.
```

File: model_cards/Rostlab/prot_t5_xl_bfd/README.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+This means it was pretrained on the raw protein sequences only, with no humans labelling them in any way (which is why it can use lots of
+publicly available data) with an automatic process to generate inputs and labels from those protein sequences.
+
+One important difference between our T5 model and the original T5 version is the denosing objective.
```
Comment (patrickvonplaten) [inline model_cards/Rostlab/prot_t5_xl_bfd/README.md:22]:
```suggestion
One important difference between this T5 model and the original T5 version is the denosing objective.
```

File: model_cards/Rostlab/prot_t5_xl_bfd/README.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+In the original T5 paper they used span denosing for the T5-3B model, in our case we used Bart like MLM denosing objective.
```
Comment (patrickvonplaten) [inline model_cards/Rostlab/prot_t5_xl_bfd/README.md:23]:
```suggestion
The original T5-3M model was pretrained using a span denosing objective, while this model was pre-trained with a Bart-like MLM denosing objective.
```

File: model_cards/Rostlab/prot_t5_xl_bfd/README.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+The masking probability follows the original T5 training with randomly masks 15% of the amino acids in the input.
```
Comment (patrickvonplaten) [inline model_cards/Rostlab/prot_t5_xl_bfd/README.md:24]:
```suggestion
The masking probability is consistent with the original T5 training by randomly masking 15% of the amino acids in the input.
```

File: model_cards/Rostlab/prot_t5_xl_bfd/README.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+ProtT5-XL-BFD is based on T5 model which pretrained on a large corpus of protein sequences in a self-supervised fashion.
+This means it was pretrained on the raw protein sequences only, with no humans labelling them in any way (which is why it can use lots of
+publicly available data) with an automatic process to generate inputs and labels from those protein sequences.
+
+One important difference between our T5 model and the original T5 version is the denosing objective.
+In the original T5 paper they used span denosing for the T5-3B model, in our case we used Bart like MLM denosing objective.
+The masking probability follows the original T5 training with randomly masks 15% of the amino acids in the input. 
+
+At the end, the feature extracted from this model revealed that the LM-embeddings from unlabeled data (only protein sequences) captured important biophysical properties governing protein
```
Comment (patrickvonplaten) [inline model_cards/Rostlab/prot_t5_xl_bfd/README.md:26]:
```suggestion
It has been shown that the features extracted from this self-supervised model (LM-embeddings) captured important biophysical properties governing protein shape.

File: model_cards/Rostlab/prot_t5_xl_bfd/README.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+shape.
+This implied learning some of the grammar of the language of life realized in protein sequences.
+
+## Intended uses & limitations
+
+The model could be used for protein feature extraction or to be fine-tuned on downstream tasks.
+We have noticed in some tasks you could gain more accuracy by fine-tuning the model rather than using it as a feature extractor.
```
Comment (patrickvonplaten) [inline model_cards/Rostlab/prot_t5_xl_bfd/README.md:33]:
```suggestion
We have noticed in some tasks on can gain more accuracy by fine-tuning the model rather than using it as a feature extractor.
```

File: model_cards/Rostlab/prot_t5_xl_bfd/README.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+We have also noticed that for feature extraction, its better to use the feature extracted from the encoder not the decoder.
```
Comment (patrickvonplaten) [inline model_cards/Rostlab/prot_t5_xl_bfd/README.md:34]:
```suggestion
We have also noticed that for feature extraction, its better to use the feature extracted from the encoder not from the decoder.
```

File: model_cards/Rostlab/prot_t5_xl_bfd/README.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+
+### How to use
+
+Here is how to use this model to get the features of a given protein sequence in PyTorch:
```
Comment (patrickvonplaten) [inline model_cards/Rostlab/prot_t5_xl_bfd/README.md:38]:
```suggestion
Here is how to use this model to extract the features of a given protein sequence in PyTorch:
```

File: model_cards/Rostlab/prot_t5_xl_bfd/README.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+The protein sequences are uppercased and tokenized using a single space and a vocabulary size of 21. The rare amino acids "U,Z,O,B" were mapped to "X".
+The inputs of the model are then of the form:
+
+```
+Protein Sequence [EOS]
+```
+
+The preprocessing step was performed on the fly, by cutting and padding the protein sequences up to 512 tokens.
+
+The details of the masking procedure for each sequence followed the original T5 model for token masking as following:
```
Comment (patrickvonplaten) [inline model_cards/Rostlab/prot_t5_xl_bfd/README.md:83]:
```suggestion
The details of the masking procedure for each sequence are as follows:
```

**Review by patrickvonplaten — APPROVED** [review]
